File: factfib.hs
fact :: Int -> Int 
fact 0 = 1 
fact n = n * fact ( n - 1 ) 

main = do 
   putStrLn "The factorial of 5 is:" 
   print (fact 5)


getNthFibo :: Int -> Int
getNthFibo 1 = 0
getNthFibo 2 = 1
getNthFibo n = getNthFibo (n-1) + getNthFibo (n-2)
main :: IO ()
main = do
putStrLn "Enter the value of n: "
nStr <- getLine
let n = read nStr :: Int
let fiboSeries = [getNthFibo x | x <- [1..n]]
putStrLn ("The first " ++ show n ++ " terms in the Fibonacci series are:")
print fiboSeries

-- Function to calculate the binomial coefficient (n choose r)
nCr :: Integer -> Integer -> Integer
nCr n r
    | r == 0 || r == n = 1
    | otherwise = nCr (n - 1) (r - 1) + nCr (n - 1) r

main :: IO ()
main = do
    putStrLn "Enter the value of n:"
    nStr <- getLine
    let n = read nStr :: Integer

    putStrLn "Enter the value of r:"
    rStr <- getLine
    let r = read rStr :: Integer

    let result = nCr n r
    putStrLn (show n ++ " C " ++ show r ++ " = " ++ show result)


-- Function to calculate the permutation (n P r)
nPr :: Integer -> Integer -> Integer
nPr n r
    | r == 0 = 1
    | otherwise = n * nPr (n - 1) (r - 1)

main :: IO ()
main = do
    putStrLn "Enter the value of n:"
    nStr <- getLine
    let n = read nStr :: Integer

    putStrLn "Enter the value of r:"
    rStr <- getLine
    let r = read rStr :: Integer

    let result = nPr n r
    putStrLn (show n ++ " P " ++ show r ++ " = " ++ show result)

File: factfib.pl
fib(0, 0).
fib(1, 1).
fib(N, Result) :-
    N > 1,
    N1 is N - 1,
    N2 is N - 2,
    fib(N1, Result1),
    fib(N2, Result2),
    Result is Result1 + Result2.



factorial(0, 1).
factorial(N, Result) :-
    N > 0,
    N1 is N - 1,
    factorial(N1, Result1),
    Result is N * Result1.



factorial(0, 1).
factorial(N, Result) :-
    N > 0,
    N1 is N - 1,
    factorial(N1, Result1),
    Result is N * Result1.
nPr(N, R, Result) :-
    N >= R,
    factorial(N, NFactorial),
    factorial(N - R, NMinusRFactorial),
    Result is NFactorial / NMinusRFactorial.



nCr(N, 0, 1). 
nCr(N, N, 1).
nCr(N, R, Result) :-
    R > 0,
    N > R,
    N1 is N - 1,
    R1 is R - 1,
    nCr(N1, R, Result1),
    nCr(N1, R1, Result2),
    Result is Result1 + Result2.

